### ISSUE TITLE
Full capability?

### ISSUE THREAD

**Issue by magnusmanske** [smalyshev/wdq2sparql#6]
For my new all-in-one tool, [PetScan](http://magnusmanske.de/wordpress/?p=385), I diss WDQ in favour of SPARQL. I would like to use WDQ2SPARQL to be able to offer live translation from WDQ syntax, but
1. I can't access WDQ2SPARQL from the petscan subdomain from JavaScript, due to cross-site access protection. Filed [a bug](https://phabricator.wikimedia.org/T131346) to enable CORS, so not hackery necessary.
2. AROUND doesn't work in SPARQL yet, but could be with the new engine, AFAIK. Needs blazegraph setup, but then it should be easy to translate from WDQ.
3. Missing features (qualifiers etc.) should be added to WDQ2SPARQL.

**Comment by magnusmanske** [comment]
Ah, apparently (1) is not (easily) possible. So could you add JSONP ("&callback=?") to wdq2sparql?

**Comment by smalyshev** [comment]
I could add jsonp, yes.

**Comment by cdrini** [comment]
Split the JSONP request into a separate issue: #8

**Comment by lokal-profil** [comment]
I did a bit on making qualifiers work (to replace WDQ queries in one of my python bots). Leaving a note here in case it might be of use for an approach in how to implement it here. I haven't checked that it works for all types of qualifiers.

So what I did was that I appended the following to the existing query (assuming main item is ?item):
```
[...]
?item p:P<main_pid> ?item_qual .
{
    ?item_qual pq:P<qual_pid> <qual_value> .
}
```
with `qual_value` varying depending on if it is CLAIM or STRING (like in a non-qualifier context).

What I did is that I reused the logic for converting CLAIM/STRING but added an `is_qualifier` flag which swapped out `wdt:` for `pq` and changes the dummy names to ensure uniqueness. That way I can re-run the main WDQ-WDQS mapping on the qualifier bits.

So `CLAIM[123]{CLAIM[123,456]}`. would give something like:
```
SELECT ?item WHERE
?item wdt:P170 ?value .
?item p:P170 ?qual0 .
{
    ?qual0 pq:P123 ?qual0_dummy0 .
    ?qual0 pq:P456 ?qual0_dummy1 .
}
```
